### ISSUE TITLE
issues with building on macOS

### ISSUE THREAD

**Issue by mkrompiec** [camfort/camfort#154]
I get the following error when trying to install on macOS, would be grateful for any advice:
```
cabal
v2-install
--jobs=8
--max-backjumps=100000
--install-method=copy
--installdir=/opt/homebrew/Cellar/camfort/1.1.2/bin

Error: cabal: Could not resolve dependencies:
[__0] trying: camfort-1.1.2 (user goal)
[__1] next goal: base (dependency of camfort)
[__1] rejecting: base-4.16.3.0/installed-4.16.3.0 (conflict: camfort =>
ghc-prim>=0.3.1.0 && <0.8, base => ghc-prim==0.8.0/installed-0.8.0)
[__1] rejecting: base-4.17.0.0, base-4.16.3.0, base-4.16.2.0, base-4.16.1.0,
base-4.16.0.0, base-4.15.1.0, base-4.15.0.0, base-4.14.3.0, base-4.14.2.0,
base-4.14.1.0, base-4.14.0.0, base-4.13.0.0, base-4.12.0.0, base-4.11.1.0,
base-4.11.0.0, base-4.10.1.0, base-4.10.0.0, base-4.9.1.0, base-4.9.0.0,
base-4.8.2.0, base-4.8.1.0, base-4.8.0.0, base-4.7.0.2, base-4.7.0.1,
base-4.7.0.0, base-4.6.0.1, base-4.6.0.0, base-4.5.1.0, base-4.5.0.0,
base-4.4.1.0, base-4.4.0.0, base-4.3.1.0, base-4.3.0.0, base-4.2.0.2,
base-4.2.0.1, base-4.2.0.0, base-4.1.0.0, base-4.0.0.0, base-3.0.3.2,
base-3.0.3.1 (constraint from non-upgradeable package requires installed
instance)
[__1] fail (backjumping, conflict set: base, camfort)
After searching the rest of the dependency tree exhaustively, these were the
goals I've had most trouble fulfilling: base, camfort

```

**Comment by raehik** [comment]
This error message is indicating (somewhat clumsily) that the ghc-prim package on your machine is too new. Could you tell me the version of GHC you're using to build CamFort? It currently supports GHC 8.2 - 8.10, but not 9.0 or 9.2. Judging by ghc-prim being 0.8.0, you should be on GHC 9.0.x.

You should be able to install GHC 8.10 using [ghcup](https://www.haskell.org/ghcup/). Or try `stack install` if you have the Stack build tool installed, and it will download a standalone GHC to use.

I'm currently working on GHC 9.0 and 9.2 support at #155 .

**Comment by dorchard** [comment]
Can we close this one now @raehik ?

**Comment by raehik** [comment]
Resolved with #155 . You should be able to build & use camfort on macOS using GHC 9.0 and 9.2 (and tentatively 9.4).

Reminds me I didn't bump the Homebrew binary package over at https://github.com/camfort/homebrew-camfort (which this issue isn't concerning). That's running through its tests currently.

**Comment by raehik** [comment]
(ping @mkrompiec - do have another go if you weren't able to resolve this before!)
